jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/solver/StageMatchTuple.java

```diff
@@ -90,24 +90,26 @@
             iterMatches = Iter.distinctAdjacent(iterMatches);
         }
 
-        // Map Tuple<NodeId> to BindingNodeId
-        Function<Tuple<NodeId>, BindingNodeId> binder = tuple -> {
-            BindingNodeId output = new BindingNodeId(input);
-            for ( int i = 0 ; i < var.length ; i++ ) {
-                Var v = var[i];
-                if ( v == null )
-                    continue;
-                NodeId id = tuple.get(i);
-                if ( reject(output, v, id) )
-                    return null;
-                output.put(v, id);
-            }
-            return output;
-        };
 
+        Function<Tuple<NodeId>, BindingNodeId> binder = tuple -> tupleToBinding(input, tuple, vars);
         return Iter.iter(iterMatches).map(binder).removeNulls();
     }
 
+    private static BindingNodeId tupleToBinding(BindingNodeId input, Tuple<NodeId> tuple, Var[] var) {
+        // Reuseable BindingNodeId builder?
+        BindingNodeId output = new BindingNodeId(input);
+        for ( int i = 0 ; i < var.length ; i++ ) {
+            Var v = var[i];
+            if ( v == null )
+                continue;
+            NodeId id = tuple.get(i);
+            if ( ! compatiable(output, v, id) )
```

Comment:
Typo - `compatiable` -> `compatible`

Reply:
These are private functions. May I put all the editorial fixes and this private renames into an ongoing separate clear-upPR I have in-progress  rather than rebase this one?

Reply:
That's fine with me